packages/schema/src/config/common.ts

```diff
@@ -50,7 +50,10 @@ export default defineUntypedSchema({
    * It is normally not needed to configure this option.
    */
   workspaceDir: {
-    $resolve: async (val, get) => val ? resolve(await get('rootDir'), val) : await findWorkspaceDir(await get('rootDir')).catch(() => get('rootDir'))
+    $resolve: async (val, get) => {
+      const rootDir = get('rootDir')
+      return val ? resolve(await rootDir, val) : await findWorkspaceDir(await rootDir).catch(() => rootDir)
```

Comment:
@danielroe is the catch supposed to return the promise? Is it not meant to be awaited?